Completer tests are failing: `ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'`

Seen in both:
- https://github.com/ipython/ipython/pull/14876
- https://github.com/ipython/ipython/pull/14877
but was passing in:
- https://github.com/ipython/ipython/pull/14874

```
FAILED tests/test_completer.py::TestCompleter::test_abspath_file_completions - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_aimport_module_completer - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_all_completions_dups - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_back_latex_completion - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_back_unicode_completion - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_cell_magics - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_class_key_completion - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_completes_attributes_in_fstring_expressions - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_completes_globals_as_args_of_methods - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_completes_in_dict_expressions - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_completion_have_signature - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_completions_have_type - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_custom_completion_error - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_custom_completion_ordering - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
FAILED tests/test_completer.py::TestCompleter::test_dataframe_key_completion - ModuleNotFoundError: No module named 'langchain_core._api.__warningregistry__'
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -77,8 +77,6 @@ test = [
 test_extra = [
     "ipython[test]",
     "curio",
-    # mask https://github.com/langchain-ai/langchain/pull/30769
-    "langchain-core<0.3.52",
     "jupyter_ai",
     "matplotlib!=3.2.0",
     "nbformat",

